### PR TITLE
feat: Allow file resource aggregation type count [DHIS2-5001]

### PR DESCRIPTION
**Pull request by d-bernat**

### CODE REVIEW

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/filter/AggregateableDataElementFilter.java [hunk 1]
```diff
@@ -33,15 +33,14 @@
 /**
  * @author Lars Helge Overland
  */
-public class AggregatableDataElementFilter
+public class AggregateableDataElementFilter
```
Comment (larshelge) [inline dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/filter/AggregateableDataElementFilter.java:36]: We called it `Aggregatable` on purpose as that seems to be the accepted English word :-) 

https://en.wiktionary.org/wiki/aggregatable

https://www.yourdictionary.com/aggregatable

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java [hunk 2]
```diff
@@ -48,30 +48,45 @@
 @JacksonXmlRootElement( localName = "valueType", namespace = DxfNamespaces.DXF_2_0 )
 public enum ValueType
 {
-    TEXT( String.class, true ),
-    LONG_TEXT( String.class, true ),
-    LETTER( String.class, true ),
+    TEXT( String.class,
+        ( aggregationType ) -> aggregationType.isAggregateable() && aggregationType == AggregationType.NONE ),
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java:52]: This logic seems redundant, AggregationType.NONE already has aggregatable set to true.

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/filter/AggregateableDataElementFilter.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/filter/AggregateableDataElementFilter.java:36]: Ok, I  found both versions, so I will change it to Aggregatable

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java]: it was removed together with lambdas removing

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java [hunk 3]
```diff
@@ -176,9 +175,31 @@ public boolean isNumeric()
...
+        {
+            return false;
+        }
+
+        if ( this == FILE_RESOURCE && aggregationType != AggregationType.COUNT )
+        {
+            return false;
+        }
+
+        if ( (this == TEXT || this == LONG_TEXT || this == LETTER) )
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java:195]: Remove inner paranthesis.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java [hunk 4]
```diff
@@ -176,9 +175,31 @@ public boolean isNumeric()
...
+        {
+            return aggregationType == AggregationType.NONE;
+        }
+        else
+        {
+            return aggregationType != AggregationType.NONE;
+        }
     }
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java:203]: Can we add a unit test to verify the behavior?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java (hunk 3 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java:195]: Done

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java (hunk 4 above)
Comment (d-bernat) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java:203]: Done

File: dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/filter/AggregateableDataElementFilter.java (hunk 1 above)
Comment (larshelge) [reply dhis-2/dhis-support/dhis-support-system/src/main/java/org/hisp/dhis/system/filter/AggregateableDataElementFilter.java]: Great :-)

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java (hunk 4 above)
Comment (larshelge) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/ValueType.java:203]: Great.

**Review by larshelge — APPROVED** [review]

**Review by maikelarabori — APPROVED** [review]

**Pull request merged**